Server/Communication/tasks.py

```diff
@@ -33,40 +34,106 @@ def is_message_expired(message):
 
 	return False
 
+def send_notification(device, message):
+	"""
+	Sends the given device a notification.
+
+	:type device: FCMDevice
+	:param message: The model of the user's phone
+
+	:type messsage: AsyncMessage
+	:param message: Contains the data about the notification settings
+	"""
+	LOGGER.debug("Sending the user a notification!")
+	notify_data = {
+		"title": message.device_title,
+		"body": message.device_message,
+		"sound": message.device_sound
+	}
+
+	if message.device_icon:
+		notify_data["icon"] = message.device_icon
+
+	result = device.send_message(**notify_data)
+	if result['success'] == 1:
+		LOGGER.debug("Notification sent successfully!")
+	else:
+		LOGGER.debug("Notification failed to be sent!")
+
+def send_websocket_message(message):
+	"""
+	Sends the user's device a message via websocket.
+
+	:type messsage: AsyncMessage
+	:param message: Contains the data about the websocket message settings
+	"""
+	LOGGER.debug("Sending the user a websocket message!")
```

Comment:
Can we add which user is being sent as part of this log as well?

Reply:
Actually, I see that message further down, can this be deleted? Doesn't seem to add much if the message is printed on line 78